### PR TITLE
Check import urls validity

### PR DESCRIPTION
**Pull request by rafatower**

### CODE REVIEW

**Comment by rafatower** [comment]
@juanignaciosl I'm fixing some of the houndy complains and will test in staging. Meanwhile, can you please review?

File: app/controllers/api/json/imports_controller.rb [hunk 1]
```diff
@@ -37,8 +39,10 @@ def create
         options = default_creation_options
 
         if params[:url].present?
+          url = params.fetch(:url)
+          valide_url!(url) unless Rails.env.development? || Rails.env.test?
```
Comment (juanignaciosl) [inline app/controllers/api/json/imports_controller.rb]: Not important, but this should be `validate_url`, I think :-) I'd use it in the previous line, it's not long.

File: lib/carto/url_validator.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+module Carto
+  module UrlValidator
+    class InvalidUrlError < StandardError
+      def initialize(msg = "Invalid URL, cannot connect dataset.")
+        super
+      end
+    end
+
+    def validate_url!(str)
+      raise InvalidUrlError unless valid_url?(str)
```
Comment (juanignaciosl) [inline lib/carto/url_validator.rb]: Adding a parameter to `InvalidUrlError` so exception message contains the url might be helpful.

File: lib/carto/url_validator.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    end
+
+    private
+
+    def valid_url?(str)
```
Comment (juanignaciosl) [inline lib/carto/url_validator.rb]: Maybe it should return `false` for `nil`, which is not a valid URL?

File: lib/carto/url_validator.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      uri = URI.parse(str)
+      if uri.is_a?(URI::HTTP) && (uri.port == 80 || uri.port == 443)
```
Comment (juanignaciosl) [inline lib/carto/url_validator.rb]: Uhm. Why hardcoding the port? That's not actually needed for a URL to be valid. If that's a check you want to do all around the code, maybe you could add a `valid_ports` parameter which defaults to `[80, 443]` (and check inclusion).

File: lib/carto/url_validator.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        return true
+      else
+        return false
```
Comment (juanignaciosl) [inline lib/carto/url_validator.rb]: This `if .. else ..` returning the condition is not needed, you can just write the condition, method will return its evaluation.

File: lib/carto/url_validator.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
+require 'uri'
+
+module Carto
+  module UrlValidator
+    class InvalidUrlError < StandardError
+      def initialize(msg = "Invalid URL, cannot connect dataset.")
+        super
```
Comment (juanignaciosl) [inline lib/carto/url_validator.rb]: I think this should be `super(msg)`, isn't it?.

File: lib/carto/url_validator.rb (hunk 2 above)
Comment (juanignaciosl) [inline lib/carto/url_validator.rb]: PS: in fact I'd hardcode the current default value for `msg`. Something like this:
```ruby
def initialize(url)
    super("Invalid URL, cannot connect dataset.: #{url}")
end
```

**Comment by juanignaciosl** [comment]
I've made some comments, most of them really minor. :+1:

File: app/controllers/api/json/imports_controller.rb (hunk 1 above)
Comment (rafatower) [inline app/controllers/api/json/imports_controller.rb]: :open_mouth: I'll fix it.

File: lib/carto/url_validator.rb (hunk 6 above)
Comment (rafatower) [inline lib/carto/url_validator.rb]: Actually it works as expected in irb. But I prefer your approach below.

File: lib/carto/url_validator.rb (hunk 5 above)
Comment (rafatower) [inline lib/carto/url_validator.rb]: correct, too much time programming in plpgsql :smile:

**Pull request merged**